src/native/s3select/s3select_napi.cpp

```diff
@@ -0,0 +1,256 @@
+/* Copyright (C) 2016 NooBaa */
+
+#include "../../../src/native/third_party/isa-l/include/crc.h"
+#include "../../../submodules/s3select/include/s3select.h"
+#include "../util/common.h"
+#include "../util/napi.h"
+#include <arpa/inet.h>
+
+namespace noobaa
+{
+
+enum InputFormat
+{
+    csv,
+    json,
+    parquet
+};
+
+class SelectWorker : public Napi::AsyncWorker
+{
+public:
+    Napi::Promise::Deferred deferred;
+    s3selectEngine::csv_object& csv_object;
+    s3selectEngine::json_object& json_object;
+    std::string select;
+    const char* buffer;
+    const unsigned int buffer_len;
+    const bool is_flush;
+    const InputFormat input_format;
+    const u_int8_t* headers_bytes;
+    const u_int32_t headers_len;
+
+    SelectWorker(Napi::Env env,
+        s3selectEngine::csv_object& csv_object,
+        s3selectEngine::json_object& json_object,
+        const char* buffer,
+        const unsigned int buffer_len,
+        const bool is_flush,
+        const InputFormat input_format,
+        const u_int8_t* headers_bytes,
+        const u_int32_t headers_len)
+        : AsyncWorker(env)
+        , csv_object(csv_object)
+        , json_object(json_object)
+        , buffer(buffer)
+        , buffer_len(buffer_len)
+        , is_flush(is_flush)
+        , input_format(input_format)
+        , deferred(env)
+        , headers_bytes(headers_bytes)
+        , headers_len(headers_len)
+    {}
+
+    /*~SelectWorker() {
```

Comment:
Was it accidentally left (to check that the object was destroyed)? @alphaprinz

Reply:
Maybe we can use a debug flag and when we set it we will have all the cpp code for debug use.